Objects/setobject.c

```diff
@@ -1419,11 +1430,16 @@ set_difference_update_internal(PySetObject *so, PyObject *other)
             Py_INCREF(other);
         }
 
-        while (set_next((PySetObject *)other, &pos, &entry))
+        while (set_next((PySetObject *)other, &pos, &entry)) {
+            PyObject *key = entry->key;
+            Py_INCREF(key);
             if (set_discard_entry(so, entry->key, entry->hash) < 0) {
                 Py_DECREF(other);
+                Py_DECREF(key);
```

Comment:
Suggesting to simplify this to:

```
Py_INCREF(key);
int rc = set_discard_entry(so, entry->key, entry->hash);
Py_DECREF(key);
if (rc < 0) {
```

Reply:
The current PR matches the `while ((key = PyIter_Next(it)) != NULL) {` loop below, so I'm incled to keep this one as is.

Reply:
np